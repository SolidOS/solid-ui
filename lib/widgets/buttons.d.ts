import { IndexedFormula, NamedNode } from 'rdflib';
export type StatusAreaContext = {
    statusArea?: HTMLElement;
    div?: HTMLElement;
    dom?: HTMLDocument;
};
export type ButtonType = 'Primary' | 'Secondary';
export type ButtonWidgetOptions = {
    buttonColor?: ButtonType;
    needsBorder?: boolean;
};
export type RenderAsDivOptions = {
    image?: HTMLImageElement;
    title?: string;
    deleteFunction?: () => void;
    link?: boolean;
    noun?: string;
    draggable?: boolean;
    clickable?: boolean;
    onClickFunction?: () => void;
    wrapInATR?: boolean;
};
/**
 * Display an error message block
 */
export declare function complain(context?: StatusAreaContext, err?: string): void;
/**
 * Remove all the children of an HTML element
 */
export declare function clearElement(ele: HTMLElement): HTMLElement;
/**
 * To figure out the log URI from the full URI used to invoke the reasoner
 */
export declare function extractLogURI(fullURI: any): any;
/**
 * By default, converts e.g. '2020-02-19T19:35:28.557Z' to '19:35'
 * if today is 19 Feb 2020, and to 'Feb 19' if not.
 * @@@ TODO This needs to be changed to local time
 * @param noTime Return a string like 'Feb 19' even if it's today.
 */
export declare function shortDate(str?: string, noTime?: boolean): string;
/**
 * Format a date and time
 * @param date for instance `new Date()`
 * @param format  for instance '{FullYear}-{Month}-{Date}T{Hours}:{Minutes}:{Seconds}.{Milliseconds}'
 * @returns for instance '2000-01-15T23:14:23.002'
 */
export declare function formatDateTime(date: Date, format: string): string;
/**
 * Get a string representation of the current time
 * @returns for instance '2000-01-15T23:14:23.002'
 */
export declare function timestamp(): string;
/**
 * Get a short string representation of the current time
 * @returns for instance '23:14:23.002'
 */
export declare function shortTime(): string;
/**
 * Sets the best name we have and looks up a better one
 */
export declare function setName(element: HTMLElement, x: NamedNode): void;
/**
 * Set of suitable images
 * See also [[findImage]]
 * @param x The thing for which we want to find an image
 * @param kb The RDF store to look in
 * @returns It goes looking for triples in `kb`,
 *          `(subject: x), (predicate: see list below) (object: image-url)`
 *          to find any image linked from the thing with one of the following
 *          predicates (in order):
 *          * ns.sioc('avatar')
 *          * ns.foaf('img')
 *          * ns.vcard('logo')
 *          * ns.vcard('hasPhoto')
 *          * ns.vcard('photo')
 *          * ns.foaf('depiction')

 */
export declare function imagesOf(x: NamedNode | null, kb: IndexedFormula): any[];
/**
 * Best logo or avatar or photo etc to represent someone or some group etc
 */
export declare const iconForClass: {
    'solid:AppProviderClass': string;
    'solid:AppProvider': string;
    'solid:Pod': string;
    'vcard:Group': string;
    'vcard:Organization': string;
    'vcard:Individual': string;
    'schema:Person': string;
    'foaf:Person': string;
    'foaf:Agent': string;
    'acl:AuthenticatedAgent': string;
    'prov:SoftwareAgent': string;
    'vcard:AddressBook': string;
    'trip:Trip': string;
    'meeting:LongChat': string;
    'meeting:Meeting': string;
    'meeting:Project': string;
    'ui:Form': string;
    'rdfs:Class': string;
    'rdf:Property': string;
    'owl:Ontology': string;
    'wf:Tracker': string;
    'wf:Task': string;
    'wf:Open': string;
    'wf:Closed': string;
};
/**
 * Find an image for this thing as a class
 */
export declare function findImageFromURI(x: NamedNode | string): string | null;
/**
 * Find something we have as explicit image data for the thing
 * See also [[imagesOf]]
 * @param thing The thing for which we want to find an image
 * @returns The URL of a globe icon if thing equals `ns.foaf('Agent')`
 *          or `ns.rdf('Resource')`. Otherwise, it goes looking for
 *          triples in `store`,
 *          `(subject: thing), (predicate: see list below) (object: image-url)`
 *          to find any image linked from the thing with one of the following
 *          predicates (in order):
 *          * ns.sioc('avatar')
 *          * ns.foaf('img')
 *          * ns.vcard('logo')
 *          * ns.vcard('hasPhoto')
 *          * ns.vcard('photo')
 *          * ns.foaf('depiction')
 */
export declare function findImage(thing: NamedNode): string;
/**
 * ToDo: Also add icons for *properties* like  home, work, email, range, domain, comment,
 */
export declare function setImage(element: HTMLElement, thing: NamedNode): void;
export declare function faviconOrDefault(dom: HTMLDocument, x: NamedNode): HTMLObjectElement | HTMLImageElement;
/**
 * Delete button with a check you really mean it
 * @@ Supress check if command key held down?
 */
export declare function deleteButtonWithCheck(dom: HTMLDocument, container: HTMLElement, noun: string, deleteFunction: () => any): HTMLImageElement;
export declare function button(dom: HTMLDocument, iconURI: string | undefined, text: string, handler?: (_event: any) => void, options?: ButtonWidgetOptions): HTMLButtonElement;
export declare function cancelButton(dom: HTMLDocument, handler: (_event?: any) => void): HTMLButtonElement;
export declare function continueButton(dom: HTMLDocument, handler: (_event: any) => void): HTMLButtonElement;
export declare function askName(dom: HTMLDocument, kb: IndexedFormula, container: HTMLDivElement, predicate?: NamedNode, theClass?: NamedNode, noun?: string): Promise<unknown>;
/**
 * A TR to represent a draggable person, etc in a list
 *
 * pred is unused param at the moment
 */
export declare const personTR: typeof renderAsRow;
export declare function renderAsRow(dom: HTMLDocument, pred: NamedNode, obj: NamedNode, options: any): HTMLTableRowElement;
export declare function createNameDiv(dom: HTMLDocument, div: HTMLDivElement, title: string | undefined, obj: NamedNode): void;
export declare function createLinkDiv(dom: HTMLDocument, div: HTMLDivElement, obj: NamedNode, options: RenderAsDivOptions): void;
/**
 * A Div to represent a draggable person, etc in a list
 * configurable to add an onClick listener
 */
export declare function renderAsDiv(dom: HTMLDocument, obj: NamedNode, options: RenderAsDivOptions): HTMLElement;
/**
 * Refresh a DOM tree recursively
 */
export declare function refreshTree(root: any): void;
/**
 * Options argument for [[attachmentList]] function
 */
export type attachmentListOptions = {
    doc?: NamedNode;
    modify?: boolean;
    promptIcon?: string;
    predicate?: NamedNode;
    uploadFolder?: NamedNode;
    noun?: string;
};
/**
 * Component that displays a list of resources, for instance
 * the attachments of a message, or the various documents related
 * to a meeting.
 * Accepts dropping URLs onto it to add attachments to it.
 */
export declare function attachmentList(dom: HTMLDocument, subject: NamedNode, div: HTMLElement, options?: attachmentListOptions): HTMLTableElement;
/**
 * Event Handler for links within solid apps.
 *
 * Note that native links have constraints in Firefox, they
 * don't work with local files for instance (2011)
 */
export declare function openHrefInOutlineMode(e: Event): void;
/**
 * Make a URI in the Tabulator.org annotation store out of the URI of the thing to be annotated.
 *
 * @@ Todo: make it a personal preference.
 */
export declare function defaultAnnotationStore(subject: NamedNode): NamedNode | undefined;
/**
 * Retrieve all RDF class URIs from solid-ui's RDF store
 * @returns an object `ret` such that `Object.keys(ret)` is
 * the list of all class URIs.
 */
export declare function allClassURIs(): {
    [uri: string]: boolean;
};
/**
 * Figuring which properties we know about
 *
 * When the user inputs an RDF property, like for a form field
 * or when specifying the relationship between two arbitrary things,
 * then er can prompt them with properties the session knows about
 *
 * TODO: Look again by catching this somewhere. (On the kb?)
 * TODO: move to diff module? Not really a button.
 * @param {Store} kb The quadstore to be searched.
 */
export declare function propertyTriage(kb: IndexedFormula): any;
/**
 * General purpose widgets
 */
/**
 * A button for jumping
 */
export declare function linkButton(dom: HTMLDocument, object: NamedNode): HTMLElement;
/**
 * A button to remove some other element from the page
 */
export declare function removeButton(dom: HTMLDocument, element: HTMLElement): HTMLButtonElement;
export declare function selectorPanel(dom: HTMLDocument, kb: IndexedFormula, type: NamedNode, predicate: NamedNode, inverse: boolean, possible: NamedNode[], options: {
    connectIcon?: string;
}, callbackFunction: (_x: NamedNode, _e: Event, _selected: boolean) => void, linkCallback: (_x: NamedNode, _e: Event, _inverse: boolean, _setStyleFunction: () => void) => void): HTMLElement;
export declare function selectorPanelRefresh(list: HTMLElement, dom: HTMLDocument, kb: IndexedFormula, type: NamedNode, predicate: NamedNode, inverse: boolean, possible: NamedNode[], options: {
    connectIcon?: string;
}, callbackFunction: (_x: NamedNode, _e: Event, _selected: boolean) => void, linkCallback: (_x: NamedNode, _e: Event, _inverse: boolean, _setStyleFunction: () => void) => void): HTMLElement;
export declare let index: any;
/**
 * Stick a stylesheet link the document if not already there
 */
export declare function addStyleSheet(dom: HTMLDocument, href: string): void;
export declare function isAudio(file?: NamedNode): boolean;
export declare function isVideo(file?: NamedNode): boolean;
/**
 *
 */
export declare function isImage(file?: NamedNode, kind?: string): boolean;
/**
 * File upload button
 * @param dom The DOM aka document
 * @param  droppedFileHandler Same handler function as drop, takes array of file objects
 * @returns {Element} - a div with a button and a inout in it
 * The input is hidden, as it is uglky - the user clicks on the nice icons and fires the input.
 */
export declare function fileUploadButtonDiv(dom: HTMLDocument, droppedFileHandler: (_files: FileList) => void): HTMLDivElement;
//# sourceMappingURL=buttons.d.ts.map