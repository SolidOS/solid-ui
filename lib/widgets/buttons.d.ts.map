{"version":3,"file":"buttons.d.ts","sourceRoot":"","sources":["../../src/widgets/buttons.ts"],"names":[],"mappings":"AAEA,OAAO,EAAE,cAAc,EAAE,SAAS,EAAsB,MAAM,QAAQ,CAAA;AAyBtE,MAAM,MAAM,iBAAiB,GAAG;IAC9B,UAAU,CAAC,EAAE,WAAW,CAAA;IACxB,GAAG,CAAC,EAAE,WAAW,CAAA;IACjB,GAAG,CAAC,EAAE,YAAY,CAAA;CACnB,CAAA;AACD,MAAM,MAAM,UAAU,GAAG,SAAS,GAAG,WAAW,CAAA;AAEhD,MAAM,MAAM,mBAAmB,GAAG;IAChC,WAAW,CAAC,EAAE,UAAU,CAAC;IACzB,WAAW,CAAC,EAAE,OAAO,CAAA;CACtB,CAAA;AAED,MAAM,MAAM,kBAAkB,GAAG;IAC/B,KAAK,CAAC,EAAE,gBAAgB,CAAC;IACzB,KAAK,CAAC,EAAE,MAAM,CAAC;IACf,cAAc,CAAC,EAAE,MAAM,IAAI,CAAC;IAC5B,IAAI,CAAC,EAAE,OAAO,CAAC;IACf,IAAI,CAAC,EAAE,MAAM,CAAC;IACd,SAAS,CAAC,EAAE,OAAO,CAAC;IACpB,SAAS,CAAC,EAAE,OAAO,CAAC;IACpB,eAAe,CAAC,EAAE,MAAM,IAAI,CAAC;IAC7B,SAAS,CAAC,EAAE,OAAO,CAAA;CACpB,CAAA;AAqBD;;GAEG;AACH,wBAAgB,QAAQ,CAAE,OAAO,CAAC,EAAE,iBAAiB,EAAE,GAAG,CAAC,EAAE,MAAM,QAMlE;AAED;;GAEG;AACH,wBAAgB,YAAY,CAAE,GAAG,EAAE,WAAW,eAK7C;AAED;;GAEG;AACH,wBAAgB,aAAa,CAAE,OAAO,KAAA,OAIrC;AAED;;;;;GAKG;AACH,wBAAgB,SAAS,CAAE,GAAG,CAAC,EAAE,MAAM,EAAE,MAAM,CAAC,EAAE,OAAO,GAAG,MAAM,CAkCjE;AAED;;;;;GAKG;AACH,wBAAgB,cAAc,CAAE,IAAI,EAAE,IAAI,EAAE,MAAM,EAAE,MAAM,GAAG,MAAM,CAYlE;AAED;;;GAGG;AACH,wBAAgB,SAAS,IAAK,MAAM,CAKnC;AAED;;;GAGG;AACH,wBAAgB,SAAS,IAAK,MAAM,CAKnC;AAID;;GAEG;AACH,wBAAgB,OAAO,CAAE,OAAO,EAAE,WAAW,EAAE,CAAC,EAAE,SAAS,QAoB1D;AAED;;;;;;;;;;;;;;;;GAgBG;AACH,wBAAgB,QAAQ,CAAE,CAAC,EAAE,SAAS,GAAG,IAAI,EAAE,EAAE,EAAE,cAAc,GAAG,GAAG,EAAE,CAQxE;AAED;;GAEG;AACH,eAAO,MAAM,YAAY;;;;;;;;;;;;;;;;;;;;;;;;;CA2BxB,CAAA;AAmBD;;GAEG;AACH,wBAAgB,gBAAgB,CAAE,CAAC,EAAE,SAAS,GAAG,MAAM,GAAG,MAAM,GAAG,IAAI,CA+BtE;AAED;;;;;;;;;;;;;;;;GAgBG;AACH,wBAAgB,SAAS,CAAE,KAAK,EAAE,SAAS,GAAG,MAAM,CAcnD;AA2CD;;GAEG;AACH,wBAAgB,QAAQ,CAAE,OAAO,EAAE,WAAW,EAAE,KAAK,EAAE,SAAS,QAsB/D;AAID,wBAAgB,gBAAgB,CAAE,GAAG,EAAE,YAAY,EAAE,CAAC,EAAE,SAAS,wCAsBhE;AA0DD;;;GAGG;AACH,wBAAgB,qBAAqB,CACnC,GAAG,EAAE,YAAY,EACjB,SAAS,EAAE,WAAW,EACtB,IAAI,EAAE,MAAM,EACZ,cAAc,EAAE,MAAM,GAAG,oBA0B1B;AAWD,wBAAgB,MAAM,CAAE,GAAG,EAAE,YAAY,EAAE,OAAO,EAAE,MAAM,GAAG,SAAS,EAAE,IAAI,EAAE,MAAM,EAClF,OAAO,CAAC,EAAE,CAAC,MAAM,EAAE,GAAG,KAAK,IAAI,EAC/B,OAAO,GAAE,mBAAoE,qBA+D9E;AASD,wBAAgB,YAAY,CAAE,GAAG,EAAE,YAAY,EAAE,OAAO,EAAE,CAAC,MAAM,CAAC,EAAE,GAAG,KAAK,IAAI,qBAM/E;AASD,wBAAgB,cAAc,CAAE,GAAG,EAAE,YAAY,EAAE,OAAO,EAAE,CAAC,MAAM,EAAE,GAAG,KAAK,IAAI,qBAEhF;AAQD,wBAAgB,OAAO,CACrB,GAAG,EAAE,YAAY,EACjB,EAAE,EAAE,cAAc,EAClB,SAAS,EAAE,cAAc,EACzB,SAAS,CAAC,EAAE,SAAS,EACrB,QAAQ,CAAC,EAAE,SAAS,EACpB,IAAI,CAAC,EAAE,MAAM,oBA0Cd;AAED;;;;GAIG;AACH,eAAO,MAAM,QAAQ,oBAAc,CAAA;AAEnC,wBAAgB,WAAW,CAAE,GAAG,EAAE,YAAY,EAAE,IAAI,EAAE,SAAS,EAAE,GAAG,EAAE,SAAS,EAAE,OAAO,EAAE,GAAG,GAAG,mBAAmB,CAwClH;AAQD,wBAAgB,aAAa,CAAE,GAAG,EAAE,YAAY,EAAE,GAAG,EAAE,cAAc,EAAE,KAAK,EAAE,MAAM,GAAG,SAAS,EAAE,GAAG,EAAE,SAAS,QAO/G;AAOD,wBAAgB,aAAa,CAAE,GAAG,EAAE,YAAY,EAAE,GAAG,EAAE,cAAc,EAAE,GAAG,EAAE,SAAS,EAAE,OAAO,EAAE,kBAAkB,QAgBjH;AACD;;;GAGG;AACH,wBAAgB,WAAW,CAAE,GAAG,EAAE,YAAY,EAAE,GAAG,EAAE,SAAS,EAAE,OAAO,EAAE,kBAAkB,GAAG,WAAW,CAoBxG;AAED;;GAEG;AACH,wBAAgB,WAAW,CAAE,IAAI,EAAE,GAAG,GAAG,IAAI,CAQ5C;AAED;;GAEG;AACH,MAAM,MAAM,qBAAqB,GAAG;IAClC,GAAG,CAAC,EAAE,SAAS,CAAA;IACf,MAAM,CAAC,EAAE,OAAO,CAAA;IAChB,UAAU,CAAC,EAAE,MAAM,CAAA;IACnB,SAAS,CAAC,EAAE,SAAS,CAAA;IACrB,YAAY,CAAC,EAAE,SAAS,CAAA;IACxB,IAAI,CAAC,EAAE,MAAM,CAAA;CACd,CAAA;AAED;;;;;GAKG;AACH,wBAAgB,cAAc,CAAE,GAAG,EAAE,YAAY,EAAE,OAAO,EAAE,SAAS,EAAE,GAAG,EAAE,WAAW,EAAE,OAAO,GAAE,qBAA0B,oBAuH3H;AAID;;;;;GAKG;AACH,wBAAgB,qBAAqB,CAAE,CAAC,EAAE,KAAK,QAoB9C;AAED;;;;GAIG;AACH,wBAAgB,sBAAsB,CAAE,OAAO,EAAC,SAAS,GAAE,SAAS,GAAG,SAAS,CAc/E;AAED;;;;GAIG;AACH,wBAAgB,YAAY,IAAK;IAAE,CAAC,GAAG,EAAE,MAAM,GAAG,OAAO,CAAA;CAAE,CAmB1D;AAED;;;;;;;;;;GAUG;AAEH,wBAAgB,cAAc,CAAE,EAAE,EAAE,cAAc,GAAG,GAAG,CAiCvD;AAED;;GAEG;AAEH;;GAEG;AACH,wBAAgB,UAAU,CAAE,GAAG,EAAE,YAAY,EAAE,MAAM,EAAE,SAAS,GAAG,WAAW,CAU7E;AAED;;GAEG;AACH,wBAAgB,YAAY,CAAE,GAAG,EAAE,YAAY,EAAE,OAAO,EAAE,WAAW,qBAQpE;AAuCD,wBAAgB,aAAa,CAC3B,GAAG,EAAE,YAAY,EACjB,EAAE,EAAE,cAAc,EAClB,IAAI,EAAE,SAAS,EACf,SAAS,EAAE,SAAS,EACpB,OAAO,EAAE,OAAO,EAChB,QAAQ,EAAE,SAAS,EAAE,EACrB,OAAO,EAAE;IAAE,WAAW,CAAC,EAAE,MAAM,CAAA;CAAE,EACjC,gBAAgB,EAAE,CAAC,EAAE,EAAE,SAAS,EAAE,EAAE,EAAE,KAAK,EAAE,SAAS,EAAE,OAAO,KAAK,IAAI,EACxE,YAAY,EAAE,CAAC,EAAE,EAAE,SAAS,EAAE,EAAE,EAAE,KAAK,EAAE,QAAQ,EAAE,OAAO,EAAE,iBAAiB,EAAE,MAAM,IAAI,KAAK,IAAI,eAcnG;AAED,wBAAgB,oBAAoB,CAClC,IAAI,EAAE,WAAW,EACjB,GAAG,EAAE,YAAY,EACjB,EAAE,EAAE,cAAc,EAClB,IAAI,EAAE,SAAS,EACf,SAAS,EAAE,SAAS,EACpB,OAAO,EAAE,OAAO,EAChB,QAAQ,EAAE,SAAS,EAAE,EACrB,OAAO,EAAE;IAAE,WAAW,CAAC,EAAE,MAAM,CAAA;CAAE,EACjC,gBAAgB,EAAE,CAAC,EAAE,EAAE,SAAS,EAAE,EAAE,EAAE,KAAK,EAAE,SAAS,EAAE,OAAO,KAAK,IAAI,EACxE,YAAY,EAAE,CAAC,EAAE,EAAE,SAAS,EAAE,EAAE,EAAE,KAAK,EAAE,QAAQ,EAAE,OAAO,EAAE,iBAAiB,EAAE,MAAM,IAAI,KAAK,IAAI,eAgFnG;AAMD,eAAO,IAAI,KAAK,EAAE,GAAQ,CAAA;AAwF1B;;GAEG;AACH,wBAAgB,aAAa,CAAE,GAAG,EAAE,YAAY,EAAE,IAAI,EAAE,MAAM,GAAG,IAAI,CAepE;AAID,wBAAgB,OAAO,CAAE,IAAI,CAAC,EAAE,SAAS,WAExC;AACD,wBAAgB,OAAO,CAAE,IAAI,CAAC,EAAE,SAAS,WAExC;AACD;;GAEG;AACH,wBAAgB,OAAO,CAAE,IAAI,CAAC,EAAE,SAAS,EAAE,IAAI,CAAC,EAAE,MAAM,GAAG,OAAO,CAkBjE;AAED;;;;;;GAMG;AAEH,wBAAgB,mBAAmB,CACjC,GAAG,EAAE,YAAY,EACjB,kBAAkB,EAAE,CAAC,MAAM,EAAE,QAAQ,KAAK,IAAI,kBAkC/C"}