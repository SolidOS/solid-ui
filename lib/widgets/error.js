"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.errorMessageBlock = errorMessageBlock;
var _widgets = require("../widgets");
var style = _interopRequireWildcard(require("../style"));
var _styleConstants = _interopRequireDefault(require("../styleConstants"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
/**
 * Create an error message block
 * @param dom The DOM on which dom.createElement will be called
 * @param err The error message string to display (or an error object)
 * @param backgroundColor Background color. Default: '#fee'
 * @param err2 Is the second param is a string, you can put the original Error in here
 * @returns A div element with the err string
 *
 * This will return a DOM element you can put in the UI as a notice for the user
 *  Meanwhile the stack is dumped to the console for the developer, so you actually know
 *  where it happened!
 */
/* eslint-disable no-console */

function errorMessageBlock(dom, err, backgroundColor, err2) {
  var div = dom.createElement('div');

  /* tslint:disable-next-line */ // Too complex for TS?
  // @ts-ignore
  var errorObject = err2 || err instanceof Error ? err : null;
  if (errorObject) {
    console.error("errorMessageBlock: ".concat(errorObject, " at: ").concat(errorObject.stack || '??'), errorObject); // @@ pick one
    div.textContent = errorObject.message;
  } else {
    div.textContent = err;
  }
  div.appendChild((0, _widgets.cancelButton)(dom, function () {
    if (div.parentNode) div.parentNode.removeChild(div);
  })).style = style.errorCancelButton;
  div.setAttribute('style', style.errorMessageBlockStyle);
  div.style.backgroundColor = backgroundColor || _styleConstants["default"].defaultErrorBackgroundColor;
  return div;
}
//# sourceMappingURL=error.js.map