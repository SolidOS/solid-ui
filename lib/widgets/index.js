"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  publicData: true
};
exports.publicData = void 0;
var _peoplePicker = require("./peoplePicker");
Object.keys(_peoplePicker).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _peoplePicker[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _peoplePicker[key];
    }
  });
});
var _dragAndDrop = require("./dragAndDrop");
Object.keys(_dragAndDrop).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _dragAndDrop[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _dragAndDrop[key];
    }
  });
});
var _buttons = require("./buttons");
Object.keys(_buttons).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _buttons[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _buttons[key];
    }
  });
});
var _iconLinks = require("./buttons/iconLinks");
Object.keys(_iconLinks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _iconLinks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _iconLinks[key];
    }
  });
});
var _error = require("./error");
Object.keys(_error).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _error[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _error[key];
    }
  });
});
var _forms = require("./forms");
Object.keys(_forms).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _forms[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _forms[key];
    }
  });
});
var _autocompleteBar = require("./forms/autocomplete/autocompleteBar");
Object.keys(_autocompleteBar).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _autocompleteBar[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _autocompleteBar[key];
    }
  });
});
var _autocompletePicker = require("./forms/autocomplete/autocompletePicker");
Object.keys(_autocompletePicker).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _autocompletePicker[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _autocompletePicker[key];
    }
  });
});
var _publicData = _interopRequireWildcard(require("./forms/autocomplete/publicData"));
exports.publicData = _publicData;
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
//# sourceMappingURL=index.js.map