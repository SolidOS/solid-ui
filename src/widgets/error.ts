/**
 * Create an error message block
 * @param dom The DOM on which dom.createElement will be called
 * @param msg The error message string to display
 * @param backgroundColor Background color. Default: '#fee'
 * @returns A div element with the msg string
 */
export function errorMessageBlock (dom: HTMLDocument, msg: string, backgroundColor?: string): HTMLDivElement {
  var div = dom.createElement('div')
  div.setAttribute(
    'style',
    'margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: ' +
      (backgroundColor || '#fee') +
      '; color:black;'
  )
  div.textContent = msg
  return div
}
