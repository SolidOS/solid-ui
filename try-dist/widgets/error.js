"use strict";

module.exports.errorMessageBlock = errorMessageBlock;

function errorMessageBlock(dom, msg, backgroundColor) {
  var div = dom.createElement('div');
  div.setAttribute('style', 'margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: ' + (backgroundColor || '#fee') + '; color:black;');
  div.textContent = msg;
  return div;
}
//# sourceMappingURL=error.js.map