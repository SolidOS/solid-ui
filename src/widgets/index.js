/**
 * General purpose utility functions used in the panes
 * oshani@csail.mit.edu
 *
 * Includes form-oriented widgets  timbl@w3.org
 *
 * sign-in sign-up widgets are in signin.js
 *
 *  Note... For pointers to posssible text-editing code, see
 *  http://stackoverflow.com/questions/6756407/what-contenteditable-editors
 */

import * as debug from '../debug'

// var aclModule = require('./acl.js')

// Each widget should ideally live in its own file.  In order to break up this
// monolithic widget index over time, we should add new widgets to the
// 'lib/widgets/' directory, and re-export them like so:
//
// (In order to avoid name collisions, it is safely assumed that modules don't
// export widgets with the same name)
var widgets = Object.assign(
  {},
  require('./peoplePicker'),
  require('./dragAndDrop'), // uploadFiles etc
  require('./error'), // UI.widgets.errorMessageBlock
  require('./buttons'),
  require('./forms')
)

module.exports = widgets
debug.log('@@@ widgets: ', widgets)

// ends
