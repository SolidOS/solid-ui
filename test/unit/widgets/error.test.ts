import { JSDOM } from 'jsdom'

import { errorMessageBlock } from '../../../src/widgets/error'

jest.mock('solid-auth-client')
const dom = new JSDOM('<!DOCTYPE html><p>Hello world</p>').window.document

describe('errorMessageBlock', () => {
  it('exists', () => {
    expect(errorMessageBlock).toBeInstanceOf(Function)
  })
  it('creates an error message div', () => {
    expect(errorMessageBlock(dom, 'my error message', undefined)).toMatchSnapshot()
  })
  it('escapes the HTML', () => {
    expect(errorMessageBlock(dom, 'my <script> kiddie </script> message', undefined)).toMatchSnapshot()
  })
  it('sets the background color', () => {
    expect(errorMessageBlock(dom, 'my colored message', '#0ff')).toMatchSnapshot()
  })
})
