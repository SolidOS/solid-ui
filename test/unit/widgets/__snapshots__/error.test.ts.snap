// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorMessageBlock creates an error message div 1`] = `
<div
  style="margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: #fee; color:black;"
>
  my error message
</div>
`;

exports[`errorMessageBlock escapes the HTML 1`] = `
<div
  style="margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: #fee; color:black;"
>
  my &lt;script&gt; kiddie &lt;/script&gt; message
</div>
`;

exports[`errorMessageBlock sets the background color 1`] = `
<div
  style="margin: 0.1em; padding: 0.5em; border: 0.05em solid gray; background-color: #0ff; color:black;"
>
  my colored message
</div>
`;
